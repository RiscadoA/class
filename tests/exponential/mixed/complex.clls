proc main() {
    cut {
        println("!a"); !a(b: choice of { |#A: send close; close |#B: !close });
        println("#A"); #A b;
        println("send b"); send b(x.
            println("close x"); close x
        );
        println("close b"); close b
        |a: ? offer of { |#A: recv wait; wait |#B: ?wait }|
        println("?a"); ?a;
        println("call a"); call a(b);
        println("case b"); case b of {
            |#A:
                println("recv b"); recv b(t);
                println("wait t"); wait t;
                println("wait b"); wait b;
                ()
            |#B:
                println("?b"); ?b;
                ()
        }
    }
};;
