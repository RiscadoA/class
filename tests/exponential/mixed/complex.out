!a
#A
send b
close b
?a
call a
case b
recv b
close x
wait t
wait b
